src/mono/mono/eventpipe/ep-rt-mono.c

```diff
@@ -3848,11 +4974,15 @@ mono_profiler_jit_chunk_destroyed (
 	if (!EventEnabledMonoProfilerJitChunkDestroyed())
 		return;
 
+	mono_profiler_fire_event_enter ();
```

Comment:
Do you think it's worthwhile to bake the calls to `fire_event_enter` / `fire_event_exit` into the `FireEtw_XYZ` methods?  Seems errorprone to write them by hand.

Reply:
This is macros generated from manifest file shared with CoreCLR so if we do something we need to special case but we also use the same macros when replaying buffered events and then you shouldn't do enter/exit. Since this is a limited number of locations and I don't expect us to add that many new events into this provider, I believe we could write them by hand for now.